src/tests/unit/tests/electron/device-connect-view/components/device-connect-port-entry.test.tsx

```diff
@@ -8,92 +8,128 @@ import { shallow } from 'enzyme';
-        const props = SetupPropsMocks(fetch, DeviceConnectState.Default, undefined);
-        const rendered = shallow(<DeviceConnectPortEntry {...props} />);
-
-        expect(rendered.state()).toMatchSnapshot();
+    describe('renders', () => {
+        const needsValidationCases = [true, false];
 
-        const onChangeHandler = rendered.find('.port-number-field').prop('onChange') as (
-            event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
-            newValue?: string,
-        ) => void;
+        it.each(needsValidationCases)('with needsValidation = %s', needsValidation => {
+            const props = {
+                needsValidation,
+            } as DeviceConnectPortEntryProps;
 
-        onChangeHandler(null, testPortNumber.toString());
+            const rendered = shallow(<DeviceConnectPortEntry {...props} />);
 
-        onConnectedMock.verifyAll();
-        expect(rendered.state()).toMatchSnapshot();
+            expect(rendered.getElement()).toMatchSnapshot();
+        });
     });
 
-    test('validate click fetch succeeds', () => {
-        const fetch: FetchScanResultsType = _ => Promise.resolve({ deviceName: 'dev', appIdentifier: 'app' } as ScanResults);
-        const props = SetupPropsMocks(fetch, DeviceConnectState.Connected, 'dev - app');
-        ValidatePortValidateClick(props);
+    describe('user interaction', () => {
+        describe('change port number', () => {
+            const portNumberCases = [testPortNumber.toString(), '', null];
+
+            it.each(portNumberCases)('handles port text = "%s"', portNumberText => {
+                updateStateCallbackMock.setup(callback => callback(DeviceConnectState.Default, undefined));
+
+                const props = {
+                    updateStateCallback: updateStateCallbackMock.object,
+                } as DeviceConnectPortEntryProps;
+
+                const rendered = shallow(<DeviceConnectPortEntry {...props} />);
+
+                expect(rendered.state()).toMatchSnapshot();
+
+                rendered.find('.port-number-field').simulate('change', null, portNumberText);
+
+                updateStateCallbackMock.verifyAll();
+
+                expect(rendered.state()).toMatchSnapshot();
+            });
+        });
+
+        describe('validate port number', () => {
+            let props: DeviceConnectPortEntryProps;
+
+            type ValidatePortTestCase = {
+                fetch: FetchScanResultsType;
+                connectState: DeviceConnectState;
+                deviceName: string;
+                description: string;
+            };
+
+            const testCases: ValidatePortTestCase[] = [
+                {
+                    fetch: () => Promise.resolve({ deviceName: 'dev', appIdentifier: 'app' } as ScanResults),
+                    connectState: DeviceConnectState.Connected,
+                    deviceName: 'dev - app',
+                    description: 'fetch succeed',
```

Comment:
Is this leftover from an earlier iteration?

Reply:
It's for the "name" of the test... I'm not 100% happy with how does this look in the code.. but I didn't find a better way to handle the test name from the object. I'm open to suggestions.

Reply:
Updated. I was able to use a table to pass the parameters to the test function.